src/Codec/Archive/Zip.hs

```diff
@@ -342,13 +352,50 @@ writeEntry opts entry = do
 #ifndef _WINDOWS
        let modes = fromIntegral $ shiftR (eExternalFileAttributes entry) 16
        when (eVersionMadeBy entry .&. 0xFF00 == 0x0300 &&
-             modes /= 0) $ setFileMode path modes
+         modes /= 0) $ setFileMode path modes
 #endif
-
   -- Note that last modified times are supported only for POSIX, not for
   -- Windows.
   setFileTimeStamp path (eLastModified entry)
 
+#ifndef _WINDOWS
+-- | Write an 'Entry' representing a symbolic link to a file.
+-- If the 'Entry' does not represent a symbolic link or
+-- the options do not contain 'OptPreserveSymbolicLinks`, this
+-- function behaves like `writeEntry`.
+writeSymbolicLinkEntry :: [ZipOption] -> Entry -> IO ()
+writeSymbolicLinkEntry opts entry = do
+  if OptPreserveSymbolicLinks `notElem` opts
```

Comment:
Should it do this by default without checking for OptPreserveSymbolicLinks while writing? If so, this potentially is a breaking behavior change.